rfcs/text/0014_notification_service.md

```diff
@@ -0,0 +1,373 @@
+
+### Threat Cloud as another pluggable storage mechanism.
+Kibana notification service collects internal notifications and shows them in the UI for both on-prem and Cloud environments.
+Kibana reuses Notification service implementation on Cloud but enhances it with additional pluggable storage of external notifications.
+![img](../images/0014/alternative_implementation.png)
+
+The main benefit of this approach is that Kibana always uses the same service in any environment.
+But in addition to the restrictions listed in the [Drawbacks](#drawbacks) section, the following are added:
+- Each retrieval of a notification list still requires waiting for a roundtrip to UNS.
+- The list of Kibana notifications stored in an instance might be out of sync with a copy stored in UNS due to network problems. 
+- Storing notifications in the Cloud deployment and their copy in UNS consumes twice the disk space.
+
+# Adoption strategy
+
+We start with introducing Kibana Notification Service as a Kibana plugin, adding more notification sources along the road.
+Integration with UNS is something that is not going to happen any time soon. The work can be divided into the following phases:
+
+### Phase I
+Kibana shows notifications created by the Newsfeed Kibana plugin in Notification flyout.
+![img](../images/0014/kns_flyout.png)
+A notification cannot have a particular recipient in the lack of a way to identify a user in the system. A notification will be shown to all the users instead.
+User-specific notification state (*read_status*) is not stored in Kibana but browser Local Storage.
+### Phase II
+When [user profiles](https://github.com/elastic/kibana/issues/17888) are supported, Kibana UI starts showing user-specific notifications.
+KIbana UI supports applying filters to the notification list.
+The very first three candidates for the integration with Notification systems:
+- Data plugin - Search sessions
+- Security solutions plugin - Cases
+- Alerting plugin - Alerts
+
+The notification system supports a user-specific state for notifications.
+Telemetry tracks the number of unique users and the number of notifications in the system.
+
+### Phase III
+Kibana UI supports pinning a notification, grouping notifications.
+Notification content supports i18n, all internal notifications are translated according to user-specific language settings.
+Load testing is done. Make adjustments according to the number of unique users: notification queue, a different schema for notification storage, etc.
+
+### Phase IV
+Kibana implements notification center UI on a separate page. 
+Notification flyout is cleaned up from "advanced" functionality.
+![img](../images/0014/notification_center.png)
+
+### Phase V
+When Unified Notification Service is supported, align on API functionality supported by both systems Kibana NS and UNS.
+Add a Kibana adapter for the Unified Notification Service.
+Refactor Kibana plugins not to emit notifications provided by UNS (Newsfeed, Suggestions).
```

Comment:
> notifications provided by UNS (Newsfeed

Does this mean Newsfeed won't work for on-premises and OSS builds?

Reply:
I'd expect we ended up with a Newsfeed plugin creating notifications for on-prem instances, but such a plugin is disabled on Cloud to allow UNS to control all the feeds.

Reply:
Gotcha! Makes sense! Thank you!